doc/concepts/embeddings.rst

```diff
@@ -0,0 +1,43 @@
+.. _embeddings:
+
+Quantum Embedding
+==================
+
+Quantum Embedding is the process of representing classical data as quantum states in a Hilbert space, i.e., *a quantum feature map*. This process is a crucial part of designing quantum algorithms and hence their complexity. For details, see [#]_ and [#]_. PennyLane provides built-in embedding templates that can do the embedding for the user; see :ref: `<pennylane.templates.embeddings>`. 
+
+Lets consider classical input data consisting of M examples, N dimensional each: :math:`\mathcal{D}=\{x^{1}, \ldots, x^{M}\}`. To embed this onto an n-bit quantum system (n qubits *or* n qumodes for discrete and continuous variable quantum computing, respectively), we can use various embedding techniques some of whcih are explained breifly below. 
+
+
+Basis Embedding
+^^^^^^^^^^^^^^^^^^^^
+
+For basis embedding, classical data has to be in the form of binary strings; each data point is an N-bit binary string :math:`x^{m}=(b_1,\ldots,b_N)` with :math:`b_i \in \{0,1\}` for :math:`i=1,\ldots,N`. Assuming all features are repesented with unit binary precision (one bit), each input example :math:`x^{m}` can be directly mapped to :math:`\mid x^{m}\rangle` quantum state. **This means that the number of quantum subsystems, n, must be at least equal to N; N features are represented by N qubits.** The input data can be reprented in the computational basis as:
+
+.. math:: \mid \mathcal{D} \rangle = \frac{1}{\sqrt{M}} \sum_{m=1}^{M} \mid x^{m} \rangle
+
+For example, lets say :math:`x^{1}=01` and :math:`x^{2}=11`. So, example1 has feature1 in the off state and feature2 in the on state, while example2 has both feature1 and feature2 in the on state. The corresponding basis encoding uses 2 qubits to represent :math:`\mid x^{1} \rangle=\mid 01 \rangle` and :math:`\mid x^{2} \rangle=\mid 11 \rangle`: 
+
+.. math:: \mid \mathcal{D} \rangle = (0) \mid 00 \rangle+\frac{1}{\sqrt{2}}\mid 01 \rangle+(0) \mid 10 \rangle+ \frac{1}{\sqrt{2}} \mid 11 \rangle
+
```

Comment:
To avoid confusion, and to not preclude how an algorithm might want to use basis embedding, I would distinguish the notion of embedding datapoints into basis states from the notion of embedding a full dataset D in superposition.

Start with the first notion of how to embed individual binary datapoints. Then move on to the idea that full datasets can be put in superposition after that.

Reply:
noted. on it!